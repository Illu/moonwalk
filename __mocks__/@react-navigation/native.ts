import { Theme, DefaultTheme as defaultTheme } from "@react-navigation/native";

export const LightTheme = {
  ...defaultTheme,
  colors: {
    ...defaultTheme.colors,
    secondary: "#000",
    background: "#f1f2f7",
    text: "#000",
    placeholderText: "#7d7f86",
    inputBackground: "#dee1e7",
    secondaryText: "#666",
    accent: "#0a84ff",
    accentBackground: "rgba(10, 132, 255, 0.1)",
    uiAccent: "#c3c4c6",
  },
};

export const DefaultTheme = LightTheme;

export const DarkTheme: Theme = LightTheme;

export const useTheme = jest.fn().mockReturnValue({
  LightTheme,
});

export const useFocusEffect = jest.fn();
export const useNavigation = jest.fn().mockReturnValue({
  navigate: jest.fn(),
  replace: jest.fn(),
  reset: jest.fn(),
  setOptions: jest.fn(),
});
export const useNavigationState = jest.fn();
export const useRoute = jest.fn().mockReturnValue({
  params: {},
});
