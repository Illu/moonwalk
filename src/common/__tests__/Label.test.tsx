import * as navigation from "@react-navigation/native";
import React from "react";

import TestRenderer from "../../helpers/testRenderer";
import { lightTheme } from "../../theme";
import Label from "../Label";

jest.spyOn(navigation, "useTheme");
navigation.useTheme.mockImplementation(() => lightTheme);

describe("Label", () => {
  it("renders correctly", () => {
    const props = {
      text: "$_LABEL_TEXT_$",
    };
    const tree = TestRenderer(<Label {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with a different color", () => {
    const props = {
      text: "$_LABEL_TEXT_$",
      color: "salmon",
    };
    const tree = TestRenderer(<Label {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
