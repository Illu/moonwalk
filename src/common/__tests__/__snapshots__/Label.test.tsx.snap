// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "borderRadius": 5,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 10,
        "paddingBottom": 2,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 2,
      },
      Object {
        "backgroundColor": "#0a84ff",
      },
      Object {},
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 12,
          "fontWeight": "bold",
        },
      ]
    }
  >
    $_LABEL_TEXT_$
  </Text>
</View>
`;

exports[`Label renders correctly with a different color 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "borderRadius": 5,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 10,
        "paddingBottom": 2,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 2,
      },
      Object {
        "backgroundColor": "salmon",
      },
      Object {},
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 12,
          "fontWeight": "bold",
        },
      ]
    }
  >
    $_LABEL_TEXT_$
  </Text>
</View>
`;
